src/compiler.rkt

```diff
@@ -219,3 +194,168 @@
 ;; Like `compile-progs`, but a single prog.
 (define (compile-prog expr ctx)
   (compose first (compile-progs (list expr) ctx)))
+
+(define (backward-pass ivec varc vregs vprecs vstart-precs)
+  (vector-fill! vprecs 0)
+  (for ([instr (in-vector ivec (- (vector-length ivec) 1) -1 -1)]
+        [n (in-range (- (vector-length vregs) 1) -1 -1)])
+
+    (define op (car instr))
+    (define tail-registers (rest instr))
+    (define srcs (map (lambda (x) (vector-ref vregs x)) tail-registers))
```

Comment:
I'd rewrite this to a `for/vector` loop, under the theory that vectors are going to be a bit faster than lists.